Feature/avx 52 output nmea gga sentence

rabad-matternet opened the pull request:
- receive NMEA GGA sentence from GGA source
- send NMEA GGA sentence to NTRIP caster/server via socket
- removed compiler warnings (that occurred in some environments)
- added comments to explain the functionality of the program
- update README

Commit:
AVX-52_output-nmea-gga-sentence
- receive NMEA GGA sentence from GGA source
- send NMEA GGA sentence to NTRIP caster/server via socket
- removed compiler warnings (that occurred in some environments)
- added comments to explain the functionality of the program

Commit:
/AVX-52_output-nmea-gga-sentence (part 2)
- updated README

rabad-matternet commented on ntripclient.c:
```diff
@@ -1636,24 +1696,60 @@ int main(int argc, char **argv)
                       if(i < 200) doloop = 0;
+
+                      // TODO: is this needed??? (write to IPC pipe back to MATE)
                       fwrite(buf, i, 1, stdout);
```
@zalice @adriennet823 @AkshayMatternet @bhiln @AydenChubbic 
This line was in the original open source version. This will send the GGA sentence (that we just got from MATE over the pipe) back to MATE (via the other pipe).  But is this necessary?  Does MATE need the GGA sent to it? I don't think it does, especially since MATE just sent this GGA sentence.

zalice replied:
MATE does not need the GGA sent back to it. I'm curious what the use-case of this was in the original implementation🤔

rabad-matternet replied:
My best guess is that the original use case was that this was actually sent to stdout as in the screen for someone to monitor.

rabad-matternet replied:
I have removed this line.

Commit:
AVX-52_output-nmea-gga-sentence (part 3)
- added usage to README

zalice commented on README:
```diff
@@ -195,3 +195,24 @@ Further Information
+
+Expected usage:
+
+./ntripclient -s <server> -m <mount point> -r <port> -u <Matternet username>
```
Is the NMEA GGA sentence required? Will passing -n followed by null function as if no GGA was specified?

rabad-matternet replied:
Is NMEA GGA sentence required? No.
Passing -n with no argument would be the same as not having the -n option.

zalice commented on README:
```diff
@@ -195,3 +195,24 @@ Further Information
+Expected usage:
+
+./ntripclient -s <server> -m <mount point> -r <port> -u <Matternet username>
+  -p <Matternet password> -n <initial NMEA GGA sentence>
```
Could we add which parameters are required, if any?

rabad-matternet replied:
Done!

zalice commented on ntripclient.c:
```diff
@@ -1636,24 +1696,60 @@ int main(int argc, char **argv)
+#ifdef NTRIPCLIENT_DEBUG_LOG
+                      fprintf(flogfile, "IPC receive %d bytes\n", i);
+                      int k = 0;
+                      for (k = 0;  k < i; k++) {
```
Can the scope of `k` be reduced to inside the for loop? e.g. declaration `for (int k = 0; ...)`

rabad-matternet replied:
Done!

zalice commented on ntripclient.c:
```diff
@@ -584,6 +590,25 @@ int main(int argc, char **argv)
+
+  int idx = 0;
+  fprintf(flogfile, "args...\n");
+  for (idx = 0; idx < argc; idx++) {
```
Can scope of `idx` be reduced to inside the for loop? e.g. declaration `for (int idx = 0; ...)`

rabad-matternet replied:
Done!

zalice reviewed (approved):
Yaaaaaas! 🔥

Commit:
AVX-52_output-nmea-gga-sentence (part 4)
- removed write of GGA sentence to stdout. In this case, stdout is the pipe to MATE.
  MATE does not need the GGA sentence.

Commit:
AVX-52_output-nmea-gga-sentence (part 5)
- added more info to README

Commit:
AVX-52_output-nmea-gga-sentence (part 6)
- changed scope of some loop indices
